src/gsy_e/gsy_e_core/user_profile_handler.py

```diff
@@ -480,7 +480,11 @@ def _read_new_datapoints_from_buffer_or_rotate_profile(
         )
 
     def rotate_profile(
-        self, profile_type: InputProfileTypes, profile, profile_uuid: str = None
+        self,
+        profile_type: InputProfileTypes,
+        profile,
+        profile_uuid: str = None,
+        input_profile_path: str = None,
```

Comment:
I agree but why the extra parameter here? If I recall correctly, this is exactly the reason why we allow `str` as the type of the `profile` parameter, in order to be able to support file paths. Did you try setting the profile path to the `profile` parameter and it did not work out?

Reply:
Please have a look into the [StrategyProfile](https://github.com/gridsingularity/gsy-e/pull/1827/files#diff-be775c2f1ca4ae57dbc0db44e06d4a7793116288106c1e06e6ade6f4793f8d52L128). If the profiles was already read once, the profile will be the profile for he first week. This will be used by the read_arbitrary_profile which will re-use it, but even worse: it will convert it again into kWh. The latter resulted in very low energy demands that are close to or lower than our Floating_point_tollerance.

I am open to better solutions here.

Reply:
I think that you can use the `input_profile` in the `StrategyProfile` class. Let me leave a possible suggestion that you can also double check if it works out.